client/checkout/woopay/email-input-iframe.js

```diff
@@ -398,7 +398,7 @@ export const handleWooPayEmailInput = async (
 				// Dispatch an event after we get the response.
 				dispatchUserExistEvent( data[ 'user-exists' ] );
 
-				if ( data[ 'user-exists' ] ) {
+				if ( data[ 'user-exists' ] && shouldOpenIframe ) {
 					openIframe( email );
 				} else if ( data.code !== 'rest_invalid_param' ) {
 					recordUserEvent( 'checkout_woopay_save_my_info_offered' );
```

Comment:
If the user exists but the Iframe is not shown are we offering the opt-in still because we're recording a track event as if we did show the opt-in

Reply:
Fixed on https://github.com/Automattic/woocommerce-payments/pull/9240/commits/284bebdd60c3923600b24965048f34f6c0b6737c.